elastic/kibana
# [Infra UI] Store asset details page state in the URL state

Closes #164147 

## Summary

This PR moves the responsibility of updating the URL state into the `<AssetDetails>` instead of providing callbacks for parent components to manage the state themselves. This way logic for updating the URL state is centralized and shared between the Host details flyout and the Host details page.

## How to test

- Run from the branch locally
- Navigate to "Hosts" screen
- Open the host details flyout
- Switch between tabs and refresh the page
- Make sure your last active tab stays active after the refresh
- Type into search fields on the "Metadata", "Processes" or "Logs" tab and refresh the page
- Make sure the search term persists
- Change the date range using the date picker and refresh the page
- Make sure that the date range persists
- All of the above should work the same for the full-page host details view
